src/libstore/derivations.cc

```diff
@@ -523,15 +522,12 @@ DrvHashModulo hashDerivationModulo(Store & store, const Derivation & drv, bool m
                 + store.printStorePath(dof.path(store, drv.name, i.first)));
             outputHashes.insert_or_assign(i.first, std::move(hash));
         }
-        return outputHashes;
-    }
-    case DerivationType::CAFloating:
-        kind = DrvHash::Kind::Deferred;
-        break;
-    case DerivationType::InputAddressed:
-        break;
-    case DerivationType::DeferredInputAddressed:
-        break;
+        return DrvHashModulo{
+            .hashes = outputHashes,
+            .kind = kind,
+        };
+    } else if (drv.type() == DerivationType::CAFloating) {
+        kind = DrvHashModulo::Kind::Deferred;
     }
```

Comment:
I realized in my PRs that this definition could move down to all be in one place. I also realized this fairly subtle thing (that we already the case) that I think deserves a comment.
```suggestion
    }
    
     /* Deferred input addressed might be done "pessimistically", so we
        still do `Regular` in that case. If it is necessarily deferred,
        we will catch it processing inputDrvs. */
    auto kind = drv.type() == DerivationType::CAFloating
        ? DrvHashModulo::Kind::Deferred
        : DrvHashModulo::Kind::Regular;
```

Reply:
Yup’ indeed that’s slightly nicer, thanks :)